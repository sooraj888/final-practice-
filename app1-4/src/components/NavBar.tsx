import { AppBar, Toolbar, Typography } from "@material-ui/core";
import React from "react";

const NavBar = (): JSX.Element => {
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography>Country Weather App</Typography>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
