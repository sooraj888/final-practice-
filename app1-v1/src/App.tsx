import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { Box } from "@material-ui/core";

function App() {
  return <Box className="App"></Box>;
}

export default App;
