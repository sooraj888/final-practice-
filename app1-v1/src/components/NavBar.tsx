import { AppBar, Toolbar, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React from "react";

const useStyles = makeStyles({
  roots: {
    fontSize: "40px",
  },
});

const NavBar = (): JSX.Element => {
  const classes = useStyles();
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography className={classes.roots}>Country Weather App</Typography>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
