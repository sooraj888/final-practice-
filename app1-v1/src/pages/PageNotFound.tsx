import { Box } from "@material-ui/core";
import React from "react";

const PageNotFound = (): JSX.Element => {
  return (
    <Box>
      <h1>Page Not Found</h1>
    </Box>
  );
};

export default PageNotFound;
