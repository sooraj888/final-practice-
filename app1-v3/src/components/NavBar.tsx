import { AppBar, Toolbar, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/styles";
import React from "react";

const useStyle = makeStyles({
  root: {
    fontSize: "40px",
  },
});

const NavBar = (): JSX.Element => {
  const classes = useStyle();
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography className={classes.root}>
          Country and Weather App
        </Typography>
      </Toolbar>
    </AppBar>
  );
};

export default NavBar;
