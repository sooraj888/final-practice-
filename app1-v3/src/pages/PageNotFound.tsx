import React from "react";

const PageNotFound = (): JSX.Element => {
  return <div>Page Not Found</div>;
};

export default PageNotFound;
