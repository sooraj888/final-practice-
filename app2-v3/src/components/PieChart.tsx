import { Box } from "@material-ui/core";
import React from "react";

type PieChartType = { percent: number };

const PieChart = ({ percent }: PieChartType): JSX.Element => {
  return (
    <div
      style={{
        display: "flex",
        width: "200px",
        height: "200px",
        background: `conic-gradient(green ${percent}% ,red 0%)`,
        borderRadius: "50%",
      }}
    ></div>
  );
};

export default PieChart;
