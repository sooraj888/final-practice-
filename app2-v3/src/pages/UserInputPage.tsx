import { Box, Button, TextField } from "@material-ui/core";
import React, { useState } from "react";
import { NavigateFunction, useNavigate } from "react-router-dom";

const UserInputPage = (): JSX.Element => {
  const [name, setName] = useState<string>("");
  const navigate: NavigateFunction = useNavigate();
  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",

        alignItems: "center",
      }}
    >
      <Box
        sx={{
          padding: "20px",
          display: "flex",
          flexDirection: "column",
          justifyContent: "space-around",
          alignItems: "center",
          width: "200px",
          height: "300px",
          border: "2px solid black",
          borderRadius: "20px",
          margin: "30px",
        }}
      >
        <TextField
          placeholder="enter name"
          required
          onChange={(e: any) => {
            setName(e.target.value);
          }}
          value={name}
          variant={"outlined"}
        ></TextField>
        <Button
          variant="contained"
          onClick={(): void => {
            navigate("/1");
          }}
        >
          Enter the Exam
        </Button>
      </Box>
    </Box>
  );
};

export default UserInputPage;
