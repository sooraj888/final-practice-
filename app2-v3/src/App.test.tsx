import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "./App";
import NavBar from "./components/NavBar";
import PieChart from "./components/PieChart";
import PageNotFound from "./pages/PageNotFound";
import UserInputPage from "./pages/UserInputPage";
import { BrowserRouter } from "react-router-dom";

test("renders learn react link", () => {
  render(<NavBar></NavBar>);
  screen.debug();
});

test("renders learn react link", () => {
  render(<NavBar></NavBar>);
  expect(screen.getByText("Question And Answer App")).toBeInTheDocument();
});

test("renders learn react link", () => {
  render(<NavBar></NavBar>);
  expect(screen.getByText("Question And Answer App")).toHaveAttribute("class");
});

test("renders le react link", () => {
  const { container } = render(<NavBar></NavBar>);
  expect(container).toMatchSnapshot();
});

test("renders learn react link", () => {
  render(<PieChart percent={50}></PieChart>);
  screen.debug();
});

test("PieChart snapshot test", () => {
  const { container } = render(<PieChart percent={50}></PieChart>);
  expect(container).toMatchSnapshot();
});

test("renders learn react link", () => {
  render(<PageNotFound></PageNotFound>);
  screen.debug();
});

test("renders learn react link", () => {
  render(<PageNotFound></PageNotFound>);
  expect(screen.getByText("Page Not Found")).toBeInTheDocument();
});

test("renders learn react link", () => {
  render(<PageNotFound></PageNotFound>);
  expect(screen.getByText("Page Not Found")).not.toHaveAttribute("class");
});

describe("", () => {
  test("PageNotFound snapshot", () => {
    const { container } = render(<PageNotFound></PageNotFound>);
    expect(container).toMatchSnapshot();
  });
});

test("UserInputPage test", () => {
  render(
    <BrowserRouter>
      <UserInputPage></UserInputPage>
    </BrowserRouter>
  );
  screen.debug();
});

test("renders learn react link", () => {
  render(
    <BrowserRouter>
      <UserInputPage></UserInputPage>
    </BrowserRouter>
  );
  expect(screen.getByText("Enter the Exam")).toBeInTheDocument();
});

test("renders learn react link", () => {
  render(
    <BrowserRouter>
      <UserInputPage></UserInputPage>
    </BrowserRouter>
  );
  expect(screen.getByText("Enter the Exam")).toHaveAttribute("class");
});

describe("", () => {
  test("UserInputPage snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <UserInputPage></UserInputPage>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});

describe("", () => {
  test("UserInputPage inserting value using fire event", () => {
    const { container } = render(
      <BrowserRouter>
        <UserInputPage></UserInputPage>
      </BrowserRouter>
    );
    const ele = screen.getByRole("textbox");
    fireEvent.change(ele, { target: { value: "sooraj" } });
    expect(ele).toHaveValue("sooraj");
  });
});
