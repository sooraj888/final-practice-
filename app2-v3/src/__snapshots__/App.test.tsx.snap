// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` PageNotFound snapshot 1`] = `
<div>
  <div>
    Page Not Found
  </div>
</div>
`;

exports[` UserInputPage snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-6"
  >
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          placeholder="enter name"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-8"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Enter the Exam
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`PageNotFound snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Question And Answer App
      </p>
    </div>
  </header>
</div>
`;

exports[`PieChart snapshot test 1`] = `
<div>
  <div
    style="display: flex; width: 200px; height: 200px; border-radius: 50%;"
  />
</div>
`;

exports[`renders le react link 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Question And Answer App
      </p>
    </div>
  </header>
</div>
`;

exports[`renders learn react link 1`] = `
<div>
  <div
    style="display: flex; width: 200px; height: 200px; border-radius: 50%;"
  />
</div>
`;
